cmd/frontend/internal/auth/userpasswd/lockout.go

```diff
@@ -111,33 +124,61 @@ func (s *lockoutStore) GenerateUnlockAccountURL(userID int32) (string, string, e
 		return "", "", err
 	}
 
-	key := strconv.Itoa(int(userID))
-
-	s.unlockToken.Set(key, []byte(tokenString))
+	s.unlockToken.SetWithTTL(key, []byte(tokenString), ttl)
 
 	path := fmt.Sprintf("/unlock-account/%s", tokenString)
 
 	return globals.ExternalURL().ResolveReference(&url.URL{Path: path}).String(), tokenString, nil
 }
 
+func formatExpiryTime(ttl int) string {
+	minutes := ttl / 60
+	seconds := ttl
+
+	if minutes < 1 {
+		return fmt.Sprintf("%d seconds", seconds)
+	}
+	return fmt.Sprintf("%d minutes", minutes)
+}
+
 func (s *lockoutStore) SendUnlockAccountEmail(ctx context.Context, userID int32, recipientEmail string) error {
-	unlockUrl, _, err := s.GenerateUnlockAccountURL(userID)
+	key := key(userID)
+	ttl, exists := s.lockouts.KeyTTL(key)
 
+	if !exists || s.UnlockEmailSent(userID) {
+		return nil
+	}
+
+	unlockUrl, _, err := s.GenerateUnlockAccountURL(userID)
 	if err != nil {
 		return err
 	}
 
-	return txemail.Send(ctx, "account_unlock", txemail.Message{
+	emailF := txemail.Send
+	if MockSendEmail != nil {
+		emailF = MockSendEmail
+	}
+	err = emailF(ctx, "account_unlock", txemail.Message{
 		To:       []string{recipientEmail},
 		Template: emailTemplates,
 		Data: struct {
 			UnlockAccountUrl string
-			ExpiryMinutes    int
+			ExpiryTime       string
 		}{
 			UnlockAccountUrl: unlockUrl,
-			ExpiryMinutes:    conf.SiteConfig().AuthUnlockAccountLinkExpiry,
+			ExpiryTime:       formatExpiryTime(ttl),
 		},
 	})
+	if err == nil {
+		s.unlockEmailSent.SetWithTTL(key, []byte("sent"), ttl)
+	}
+
+	return err
+}
```

Comment:
This is Go, baby, where `if err != nil` rules!

```suggestion
	if err != nil {
		return err
	}
	s.unlockEmailSent.SetWithTTL(key, []byte("sent"), ttl)
}
```

Reply:
I mean, both things do the same, but I understand the convention wins here :)